### ISSUE TITLE
unsolicited debug messages on stderr/syslog

### ISSUE THREAD

**Issue by jnweiger** [owncloud/client#7453]
tested with 2.6.0rc1 on Debian10

`owncloud`
  * debug messages appear on stderr

`owncloud --logfile -`
  * debug messages appear on stdout

`owncloud --logfile /dev/null`
  * silent.

This is only seen on debian 10. E.g. ubuntu 18.04 is silent per default.

**Comment by ogoffart** [comment]
what debug messages?

I guess it is either things that are not part of owncloud, or maybe they appear before we install the hook.

**Comment by jnweiger** [comment]
Basically everything. A full log output appears either on stderr, or when started with gnome launcher on syslog -- for details see
https://central.owncloud.org/t/client-on-ubuntu-18-04-spams-syslog/23158/5

**Comment by jnweiger** [comment]
My first guess: We hit the `else` branch way too often in this code snippet:
```
diff --git a/src/libsync/logger.cpp b/src/libsync/logger.cpp
index 4b4791bd7..752fa0fd1 100644
--- a/src/libsync/logger.cpp
+++ b/src/libsync/logger.cpp
@@ -40,7 +40,7 @@ static void mirallLogCatcher(QtMsgType type, const QMessageLogContext &ctx, cons
     if (!logger->isNoop()) {
         logger->doLog(qFormatLogMessage(type, ctx, message));
     } else if(type >= QtCriticalMsg) {
-        std::cerr << qPrintable(qFormatLogMessage(type, ctx, message)) << std::endl;
+        std::cerr << qPrintable(qFormatLogMessage(type, ctx, message)) << std::endl << std::endl << std::endl;
     }
 
 #if defined(Q_OS_WIN)
```
The extra newlines added here are for verifying only: they are visible on stderr now.

**Comment by michaelstingl** [comment]
Linux only?

**Comment by TheOneRing** [comment]
I'm sorry I assumed the Qt*Msg enum to be ordered but it isn't...

**Comment by michaelstingl** [comment]
Fixed in next 2.6.x
